src/layout.ts

```diff
@@ -88,6 +89,30 @@ export class TextLayoutEngine {
     };
   }
 
+   /**
+   * Check if a text segment is a heading style
+   */
+  private isHeadingStyle(segment: TextSegment, measurement: { height: number }): boolean {
+    const size = measurement.height;
+
+    return segment.fontWeight === 'bold' && inverseHeadingSizes[segment.fontSize ?? 0] !== undefined;
+  }
+
+  /**
+   * Check if a text segment is unstyled (No HTML Tag applied)
+   */
+  private isUnstyledSegment(segment: TextSegment): boolean {
+    const color = segment.color?.toLowerCase();
+
+    return (
+      segment.fontWeight === 'normal' &&
+      segment.fontStyle === 'normal' &&
+      segment.textDecoration === 'none' &&
+      (!color || color === '#000' || color === '#000000')
```

Comment:
For '#000' /'#000000' encode it to black color in colorMap and use it here. For dark theme, we can replace this with '#fff' later.